### PR TITLE
feat: add the capability to return data from duckdb and datafusion

### PR DESCRIPTION
**Pull request by EpsilonPrime**
Also includes an improved configuration system and adds a Substrait validation test.

### CODE REVIEW

File: src/gateway/converter/conversion_options.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    use_project_emit_workaround2: bool
+    use_emits_instead_of_direct: bool
+
+    return_names_with_types: bool
+
+    def __init__(self, backend: BackendOptions = None):
+        self.use_named_table_workaround = False
+        self.needs_scheme_in_path_uris = False
+        self.use_project_emit_workaround = False
+        self.use_project_emit_workaround2 = False
```
Comment (westonpace) [inline src/gateway/converter/conversion_options.py:23]: Heh, this is definitely something that will need documentation when ready for primetime.

**Review by westonpace — COMMENTED** [review]
How exactly are these backends used?  I thought the intent of the service was to translate spark connect to substrait.  Are you actually executing plans too?

File: src/gateway/converter/conversion_options.py (hunk 1 above)
Comment (EpsilonPrime) [reply src/gateway/converter/conversion_options.py:23]: Yes, I need to figure out what's going on between my expectation of project and what the backends are actually doing first.  I am going to put together some project tests in consumer testing to wrap my head around it.

**Comment by EpsilonPrime** [comment]
> How exactly are these backends used? I thought the intent of the service was to translate spark connect to substrait. Are you actually executing plans too?

Yes!  This is actually executing the plans and returning the results.  The library will convert the SparkConnect plans into Substrait ones but the interface of the service is the same as Spark.  That means that when this service is complete it could be used (with a suitable backend) as a drop-in replacement for Spark.

**Pull request merged**